src/GoTrueClient.ts

```diff
@@ -365,14 +368,35 @@ export default class GoTrueClient {
    */
   async signInWithOAuth(credentials: SignInWithOAuthCredentials): Promise<OAuthResponse> {
     await this._removeSession()
-    return this._handleProviderSignIn(credentials.provider, {
+
+    return await this._handleProviderSignIn(credentials.provider, {
       redirectTo: credentials.options?.redirectTo,
       scopes: credentials.options?.scopes,
       queryParams: credentials.options?.queryParams,
       skipBrowserRedirect: credentials.options?.skipBrowserRedirect,
+      flowType: credentials.options?.flowType,
     })
   }
 
+  /**
+   * Log in an existing user via a third-party provider.
+   */
+  async exchangeAuthCode(authCode: string): Promise<AuthResponse> {
+    // Fetched local hashed verifier
+    let codeVerifier = localStorage.getItem('pkce')
+    // let codeVerifier = await getItemAsync(this.storage, 'pkce')
```

Comment:
Left as a comment until we decide on what's a reasonable storage format. There was some mention of moving away from `localStorage` but I am unsure what the consensus is at the moment